File: design-probes/dynamic-defaults/flows/dispatcher/flow.ts
#!/usr/bin/env bun

// DESIGN PROBE ONLY: complete pseudocode against the future portable SDK.
import {
  type JsonValue,
  type RunResult,
  serveRun,
} from "@flowmd/sdk";

interface DispatchInput {
  readonly intent: string;
  readonly text: string;
}

const transformedValue = (value: JsonValue): string | undefined => {
  if (value === null || Array.isArray(value) || typeof value !== "object") {
    return undefined;
  }

  const candidate = (value as { readonly [name: string]: JsonValue }).value;
  return typeof candidate === "string" ? candidate : undefined;
};

serveRun<DispatchInput>(async (run): Promise<RunResult> => {
  const child = await run.flows.call({
    operationId: "delegate-text",
    slot: "delegate",
    intent: run.input.intent,
    input: { text: run.input.text },
  });

  const value = child.outcome === "done"
    ? transformedValue(child.output)
    : undefined;

  return value === undefined
    ? { outcome: "blocked", output: {} }
    : { outcome: "done", output: { value } };
});

File: design-probes/cordis-timer/flows/wait-on-cordis/flow.py
#!/usr/bin/env python3

# DESIGN PROBE ONLY: future Python projection of FLOW Run/1.
from flowmd_sdk import serve_run


async def wait_on_cordis(run):
    result = await run.effects.call(
        operation_id="wait-on-cordis",
        slot="delay",
        method="wait",
        input=run.input,
    )
    return {"outcome": "done", "output": result}


serve_run(wait_on_cordis)
